### PR TITLE
Working on challenges workflow and Event Participant Challenge.

### PR DESCRIPTION
**Pull request by tyte**
Some changes made to challenges. Should be tested and perhaps reworked before pushing to main.

### CODE REVIEW

File: lib/utils/model/joined_challenge/joined_challenge_school_gsa/joined_challenge_school_gsa_class.dart [hunk 1]
```diff
@@ -6,10 +6,10 @@ part 'joined_challenge_school_gsa_class.g.dart';
 @JsonSerializable(explicitToJson: true)
 class JoinedChallengeSchoolGsa {
   JoinedChallengeSchoolGsa(
-      {required this.uuid, required this.mainJoinedChallenge});
+      {required this.uuid, required this.main_joinedChallenge});
 
   String uuid;
-  JoinedChallenge mainJoinedChallenge;
+  JoinedChallenge main_joinedChallenge;
```
Comment (tyte) [inline lib/utils/model/joined_challenge/joined_challenge_school_gsa/joined_challenge_school_gsa_class.dart:12]: I think it should be "main_joined_challenge", not "main_joinedChallenge" will check it.

Comment (ignasbagdonas) [reply lib/utils/model/joined_challenge/joined_challenge_school_gsa/joined_challenge_school_gsa_class.dart:12]: Flutter convention is for files: lowercase and underscores, but in code should be used camelCase, so variable should be named **mainJoinedChallenge;**

File: lib/utils/model/challenge/challenge_story/challenge_story_class.dart [hunk 2]
```diff
@@ -5,10 +5,10 @@ part 'challenge_story_class.g.dart';
 // explicitToJson lets using nested class Challenge
 @JsonSerializable(explicitToJson: true)
 class ChallengeStory {
-  ChallengeStory({required this.uuid, required this.mainChallenge});
+  ChallengeStory({required this.uuid, required this.main_challenge});
 
   String uuid;
-  Challenge mainChallenge;
+  Challenge main_challenge;
```
Comment (ignasbagdonas) [inline lib/utils/model/challenge/challenge_story/challenge_story_class.dart:11]: Convention error. Please rename it everywhere.

File: lib/localization/app_en.arb [hunk 3]
```diff
@@ -23,5 +23,7 @@
     "challenge_type_story": "Story",
     "challenge_type_support": "Support",
     "action_join_challenge": "Join the challenge",
-    "joined_challenge_event_field_qr_code": "QR code"
+    "joined_challenge_event_field_qr_code": "Scan QR code", 
+    "joined_challenge_event_field_qr_code_wrong": "Atsiprašome, neteisingas QR kodas",
+    "joined_challenge_event_field_qr_code_right": "Užduotis atlikta"
```
Comment (karina-klinkeviciute) [inline lib/localization/app_en.arb:28]: this is English localization, so probably should be in English

**Review by karina-klinkeviciute — APPROVED** [review]

**Pull request merged**